### PR TITLE
Fix inability to interrupt timeline flick-scrolls by other means

### PR DESCRIPTION
**Pull request by peppy**
After flick-dragging the editor timeline, it should be possible to interrupt the long-running seek (based on scroll velocity) by other means, such as scrolling the mouse wheel in the playfield or double-clicking a hitobject.

This regressed with the introduction of smooth seeking in the `EditorClock` behaviour, as the local logic to ensure an external operation has not occurred was not considering a future seek that is hidden within a pending transform.

Closes the second half of #11497.

### CODE REVIEW

File: osu.Game/Screens/Edit/EditorClock.cs [hunk 1]
```diff
@@ -239,20 +263,10 @@ private void updateSeekingState()
                 // we are either running a seek tween or doing an immediate seek.
                 // in the case of an immediate seek the seeking bool will be set to false after one update.
                 // this allows for silencing hit sounds and the likes.
-                seekingOrStopped.Value = Transforms.Any();
+                seekingOrStopped.Value = IsSeeking;
```
Comment (bdach) [inline osu.Game/Screens/Edit/EditorClock.cs:266]: Thoughts about splitting `seekingOrStopped` into two `seeking` and `stopped` bindables, and then substituting what is the current `seekingOrStopped` with an `AggregateBindable`? I guess this is not *horrible* as it is right now but I still find this song and dance about having to sorta track both simultaneously a little teeth-pain-inducing.

File: osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs [hunk 2]
```diff
@@ -146,12 +146,14 @@ protected override void UpdateAfterChildren()
...
             {
-                // The track isn't running. There are two cases we have to be wary of:
-                // 1) The user flick-drags on this timeline: We want the track to follow us
-                // 2) The user changes the track time through some other means (scrolling in the editor or overview timeline): We want to follow the track time
+                // The track isn't running. There are three cases we have to be wary of:
+                // 1) The user flick-drags on this timeline and we are applying an interpolated seek on the clock, until interrupted by 2 or 3.
+                // 2) The user changes the track time through some other means (scrolling in the editor or overview timeline; clicking a hitobject etc.). We want the timeline to track the clock's time.
+                // 3) An ongoing seek transform is running from an external seek. We want the timeline to track the clock's time.
 
                 // The simplest way to cover both cases is by checking whether the scroll position has changed and the audio hasn't been changed externally
```
Comment (bdach) [inline osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs:154]:
```suggestion
                // The simplest way to cover the first two cases is by checking whether the scroll position has changed and the audio hasn't been changed externally
```

File: osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs [hunk 3]
```diff
@@ -146,12 +146,14 @@ protected override void UpdateAfterChildren()
...
-                if (Current != lastScrollPosition && editorClock.CurrentTime == lastTrackTime)
+                // Checking IsSeeking covers the third case, where the transform may not have been applied yet.
+                if (Current != lastScrollPosition && editorClock.CurrentTime == lastTrackTime && !editorClock.IsSeeking)
```
Comment (bdach) [inline osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs:156]: I *think* I can make sense of this condition, although it is getting a bit dense to parse even with the preceding comment...

File: osu.Game/Screens/Edit/EditorClock.cs (hunk 1 above)
Comment (bdach) [reply osu.Game/Screens/Edit/EditorClock.cs:266]: alternative thought would be to have a tri-state enum (seeking, stopped and playing seem like mutually exclusive states)

**Review by smoogipoo — APPROVED** [review]
Let's get this in for now since it behaves fine, and do more refactorings afterwards.

**Pull request merged**